qa/integration-tests/src/test/java/io/camunda/zeebe/it/network/LargeMessageSizeTest.java

```diff
@@ -134,4 +151,64 @@ public void shouldCompleteJobWithLargeVariables() {
     // then
     ZeebeAssertHelper.assertProcessInstanceCompleted(processInstanceEvent.getProcessInstanceKey());
   }
+
+  @Test
+  public void shouldActivateJobsByRespectingMaxMessageSize() {
+    // given
+    final var modelInstance =
+        Bpmn.createExecutableProcess("foo")
+            .startEvent()
+            .serviceTask()
+            .zeebeJobType("foo")
+            .endEvent()
+            .done();
+
+    CLIENT_RULE
+        .getClient()
+        .newDeployResourceCommand()
+        .addProcessModel(modelInstance, "foo.bpmn")
+        .send()
+        .join();
+
+    final var byteArray = new byte[1024 * 1024]; // 1 MB
+    new Random().nextBytes(byteArray);
+    final var message = new String(byteArray, StandardCharsets.UTF_8);
+
+    final int numberOfJobsToActivate = 5;
+    for (int i = 0; i < numberOfJobsToActivate; i++) {
+      final ProcessInstanceEvent event =
+          CLIENT_RULE
+              .getClient()
+              .newCreateInstanceCommand()
+              .bpmnProcessId("foo")
+              .latestVersion()
+              .variables(Map.of("message_content", message))
+              .send()
+              .join();
+
+      RecordingExporter.processInstanceRecords()
+          .withProcessInstanceKey(event.getProcessInstanceKey())
+          .withElementType(BpmnElementType.PROCESS)
+          .await();
+    }
+
+    // when
+    final JobWorkerBuilderStep3 builder =
+        CLIENT_RULE.getClient().newWorker().jobType("foo").handler(COMPLETING_JOB_HANDLER);
```

Comment:
❌ Before opening the job worker, we must ensure that all 5 jobs exist. Otherwise, the job worker might activate jobs and receive less than 5 jobs.

Reply:
Yeah, makes sense! Updated.

Reply:
❌ This is still missing in both this test and the one above